File: cipher-codes/VernamCipher.java
import java.util.Scanner;

public class VernamCipher {
    public static String encrypt(String plaintext, String key) {
        if (plaintext.length() != key.length()) {
            throw new IllegalArgumentException("The length of the key must be equal to the length of the plaintext.");
        }

        StringBuilder ciphertext = new StringBuilder();
        for (int i = 0; i < plaintext.length(); i++) {
            char p = plaintext.charAt(i);
            char k = key.charAt(i);
            char cipherChar = (char) (p ^ k);
            ciphertext.append(cipherChar);
        }

        return ciphertext.toString();
    }
    public static String decrypt(String ciphertext, String key) {
        if (ciphertext.length() != key.length()) {
            throw new IllegalArgumentException("The length of the key must be equal to the length of the ciphertext.");
        }

        StringBuilder plaintext = new StringBuilder();
        for (int i = 0; i < ciphertext.length(); i++) {
            char c = ciphertext.charAt(i);
            char k = key.charAt(i);
            char plainChar = (char) (c ^ k);
            plaintext.append(plainChar);
        }

        return plaintext.toString();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Do you want to encrypt or decrypt? (e/d): ");
        String mode = scanner.nextLine().trim().toLowerCase();

        System.out.print("Enter the plaintext or ciphertext: ");
        String text = scanner.nextLine();

        System.out.print("Enter the key: ");
        String key = scanner.nextLine();

        if (mode.equals("e")) {
            String encryptedText = encrypt(text, key);
            System.out.println("Encrypted text: " + encryptedText);
        } else if (mode.equals("d")) {
            String decryptedText = decrypt(text, key);
            System.out.println("Decrypted text: " + decryptedText);
        } else {
            System.out.println("Invalid option. Please choose 'e' to encrypt or 'd' to decrypt.");
        }

        scanner.close();
    }
}

File: cipher-codes/VernamCipher.py
def encrypt(plaintext, key):
    if len(plaintext) != len(key):
        raise ValueError("The length of the key must be equal to the length of the plaintext.")

    ciphertext = []
    for p, k in zip(plaintext, key):
        cipher_char = chr(ord(p) ^ ord(k))
        ciphertext.append(cipher_char)
    
    return ''.join(ciphertext)

def decrypt(ciphertext, key):
    if len(ciphertext) != len(key):
        raise ValueError("The length of the key must be equal to the length of the ciphertext.")
    
    plaintext = []
    for c, k in zip(ciphertext, key):
        plain_char = chr(ord(c) ^ ord(k))
        plaintext.append(plain_char)
    
    return ''.join(plaintext)

mode = input("Do you want to encrypt or decrypt? (e/d): ").strip().lower()
text = input("Enter the plaintext or ciphertext: ")
key = input("Enter the key: ")

if mode == 'e':
    encrypted_text = encrypt(text, key)
    print("Encrypted text:", encrypted_text)
elif mode == 'd':
    decrypted_text = decrypt(text, key)
    print("Decrypted text:", decrypted_text)
else:
    print("Invalid option. Please choose 'e' to encrypt or 'd' to decrypt.")